test/e2e/tests/018-surveyImport.js

```diff
@@ -43,6 +126,193 @@ describe('Survey import', () => {
+
+    await expectExists({ text: 'Export' })
+    await click('Export', toLeftOf('Delete'))
+  })
+
+  test('Unzip file', async () => {
+    await extractZip(surveyZipPath, extractedPath)
+    await expect(fs.existsSync(extractedPath)).toBeTruthy()
+    await expect(fs.existsSync(surveyExtractedPath)).toBeTruthy()
+  })
+
+  test('Check survey.json', async () => {
+    await expect(fs.existsSync(path.join(surveyExtractedPath, 'survey.json'))).toBeTruthy()
+    const survey = getSurvey()
+    await expect(Survey.getName(Survey.getSurveyInfo(survey))).toBe(surveyName)
+    await expect(Survey.getLabels(Survey.getSurveyInfo(survey))).toMatchObject({
+      en: 'Survey',
+    })
+    const languages = Survey.getLanguages(Survey.getSurveyInfo(survey))
+    await expect(languages.sort()).toEqual(['en', 'fr'].sort())
+  })
+
+  test('Check users in imported survey', async () => {
+    const users = await checkFileAndGetContent({
+      filePath: path.join(surveyExtractedPath, 'users', 'users.json'),
+    })
+
+    const usersAsArray = Object.values(users)
+
+    await expect(usersAsArray.length).toBe(1)
+
+    const tester = usersAsArray.find((_user) => User.getEmail(_user) === 'test@arena.com')
+    await expect(tester).toBeTruthy()
+  })
+
+  /* test('Remove files', async () => {
+    if (fs.existsSync(path.join(downloadPath, 'extracted'))) {
+      fs.rmdirSync(path.join(downloadPath, 'extracted'), { recursive: true })
+    }
+
+    await expect(fs.existsSync(path.join(downloadPath, 'extracted'))).not.toBeTruthy()
+  }) */
+
+  test('Check taxonomies', async () => {
+    const taxonomies = await checkFileAndGetContent({
+      filePath: path.join(surveyExtractedPath, 'taxonomies', 'taxonomies.json'),
+    })
+
+    const taxonomiesAsArray = Object.values(taxonomies)
+
+    await expect(taxonomiesAsArray.length).toBe(1)
+
+    const taxonomyUuid = Chain.getUuid(taxonomiesAsArray[0])
+    await expect(Taxonomy.getName(taxonomiesAsArray[0])).toBe('tree_species')
+    await expect(Taxonomy.getDescription('en')(taxonomiesAsArray[0])).toBe('Tree Species List')
+
+    const taxonomy = await checkFileAndGetContent({
+      filePath: path.join(surveyExtractedPath, 'taxonomies', `${taxonomyUuid}.json`),
+    })
+
+    const taxonomyMockData = fs.readFileSync(
+      path.resolve(__dirname, '..', 'resources', 'taxonomies', 'species list valid with predefined.csv')
+    )
+    const taxonomyMockDataParsed = CSVReaderSync(taxonomyMockData, { columns: true, skip_empty_lines: true })
+
+    const taxonomyMockDataParsedByCode = taxonomyMockDataParsed.reduce(
+      (acc, taxon) => ({ ...acc, [taxon.code]: { ...taxon } }),
+      {}
+    )
+    const taxonomyTaxaByCode = taxonomy.reduce((acc, taxon) => ({ ...acc, [Taxon.getCode(taxon)]: { ...taxon } }), {})
+
+    await expect(Object.keys(taxonomyMockDataParsedByCode).sort()).toEqual(Object.keys(taxonomyTaxaByCode).sort())
+
+    await taxonomy.reduce(async (promise, taxon) => {
```

Comment:
would `await PromiseUtils.each(taxonomy, async taxon => {...})` be more readable?